sharedbuffers/mapped_struct.py

```diff
@@ -81,13 +82,19 @@
 import ctypes
 from datetime import timedelta, datetime, date
 from decimal import Decimal
+from six import itervalues, reraise, iteritems, iterkeys
+from six.moves import cPickle, range
```

Comment:
importing range may make cython's loop optimizations go bad.

Check cython --annotate

Reply:
oops, yes. i'll look into it

Reply:
We could just use `range` without importing, although in py2 without cython it would be slower than xrange of course (py2 + cython is ok with `range`). I don't see another way around this.

Reply:
You can trust cython to properly implement xrange, I think.

Keep unapologetically using xrange, cython will generate proper code for it even if it's not there in py3.

Cython is not optional after all, lets remember that. The fact that tests are failing in pure python is a sign of that.

Reply:
ping. This is breaking cython optimizations. Really needs fixing.